### PR TITLE
Cofense Vision 100 release

### PR DESCRIPTION
**Pull request by crestdatasystems**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
- Search for suspicious and malicious emails.
- Create and approve jobs against emails to quarantine threats in mailboxes.
- Create quarantine jobs to remove suspicious emails from mailboxes using automation.
- Download original emails and get suspicious attachments for additional analysis.
- Add IOCs to be used in searching and quarantine job operations.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [x] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Review by idovandijk — CHANGES_REQUESTED** [review]
Thank you for waiting for my review. While Dor is reviewing your code, I reviewed your playbooks.
Excellent job! I'm really happy to have another contribution for an email security product.
Let me add my comments below:

### Message Quarantine - Cofense Vision
Very good job with the inputs and the thoroughly explained descriptions, this allows more customizability for the end user.
1. Please also add outputs for the playbook to make it possible for other playbooks using it as a subplaybook to know what search results were found.
2. I assume that the search does not complete instantaneously. If so, the `cofense-message-search-results-get` command might give incomplete results. Correct? If not familiar with your code, but if I'm correct then you should make sure your code implements **polling.** If you haven't already implemented polling, you can ask @dorschw for assistance with that as he is reviewing your code. He will gladly help.
3. What if the search yields no results? Then the `Set` command would fail. I see that you've overcome that by using a fallback for an empty string. We have a script exactly for that case - please use `SetAndHandleEmpty` which would not fail if no value is empty.
4. Suggestion (up to you to decide if you want to implement it): allow getting message content and quarantining the message to be automatic through a playbook input.
5. Suggestion: I noticed that some of the command arguments require comma-separated lists. It would be wise to support list inputs as well.

### Add IOCs - Cofense Vision
Nice.

Also are you sure you need both scripts? Why is there a dict of lists in the response from Cofense?
Kindly go over my comments and make fixes where needed or let me know if there's any question or concern.

Finally could we schedule a short demo? Dor may want to see outputs from the integration commands and I would like to see incidents showing the playbook execution. After the fixes of course.
Thanks again for your time and effort with this PR.

**Comment by idovandijk** [comment]
@crestdatasystems what time-zone are you in?

**Comment by crestdatasystems** [comment]
> e are you in? >> we are in Indian Time Zone (IST)

**Comment by crestdatasystems** [comment]
> @crestdatasystems what time-zone are you in?

We are in Indian Standard Time Zone (IST) UTC+5.30 , You can schedule the demo on Thursday/Friday or Monday, as Dor has asked to change a few things.

**Comment by idovandijk** [comment]
> > @crestdatasystems what time-zone are you in?
> 
> We are in Indian Standard Time Zone (IST) UTC+5.30 , You can schedule the demo on Thursday/Friday or Monday, as Dor has asked to change a few things.

I've sent a Zoom invite via email for tomorrow (Monday). Please let me know if that time works for you. Thank you.

File: Packs/CofenseVision/Integrations/CofenseVision/CofenseVision.py [hunk 1]
```diff
@@ -0,0 +1,2470 @@
...
+            sort_by = sort_by if sort_by else "None"
+            raise ValueError(ERROR_MESSAGE['INVALID_FORMAT'].format(sort_by, 'sort',
+                                                                    SUPPORTED_SORT_FORMAT_FOR_IOCS_LIST
+                                                                    if command == "iocs_list"
+                                                                    else SUPPORTED_SORT_FORMAT))
+
+        property_name, sort_order = sort_by.split(':')
+
+        # Checking whether the property name and sort order values are correct or not.
+        if property_name[0].lower() + property_name[1:] not in SUPPORTED_SORT[command]:
```
Comment (dorschw) [inline Packs/CofenseVision/Integrations/CofenseVision/CofenseVision.py:740]:
```suggestion
        if (property_name[0].lower() + property_name[1:]) not in SUPPORTED_SORT[command]:
```

**Review by dorschw — APPROVED** [review]

**Pull request merged**